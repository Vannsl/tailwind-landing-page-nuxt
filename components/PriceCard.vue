<template>
  <div class="price-card" :class="{ highlighted: highlighted }">
    <div class="price-content">
      <div class="price-name">{{ name }}</div>
      <div class="seperator gradient" v-if="highlighted" />
      <ul>
        <li v-for="(item, index) in features" :key="index">{{ item }}</li>
      </ul>
    </div>
    <div class="pricing">
      <div class="price">
        {{ price }}
        <span>{{ priceLabel }}</span>
      </div>
      <div class="price-action">
        <button class="gradient" ;>Sign Up</button>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'PriceCard',
  props: {
    highlighted: {
      type: Boolean,
      default: false,
      required: false
    },
    name: {
      type: String,
      required: true
    },
    price: {
      type: String,
      required: true
    },
    priceLabel: {
      type: String,
      default: '/ per user'
    },
    features: {
      type: Array,
      required: true
    }
  }
}
</script>

<style scoped>
.price-card {
  @apply flex flex-col w-5/6 mx-auto rounded-none bg-white mt-4;
}

.price-card.highlighted {
  @apply shadow-lg z-10;
}

@screen sm {
  .price-card.highlighted {
    @apply -mt-6;
  }
}

@screen lg {
  .price-card {
    @apply w-1/4 mx-0 rounded-lg;
  }

  .price-card.highlighted {
    @apply w-1/3;
  }
}

.price-card.highlighted .price-content {
  @apply text-gray-800;
}

.price-content {
  @apply flex-1 bg-white text-gray-600 rounded-t rounded-b-none overflow-hidden shadow;
}

.price-card.highlighted .price-content ul {
  @apply font-bold;
}

.price-content ul {
  @apply w-full text-center text-sm;
}

.price-card.highlighted .price-content ul li {
  @apply text-base;
}

.price-content ul li {
  @apply border-b py-4 text-sm;
}

.seperator {
  @apply h-1 w-full my-0 py-0 rounded-t;
}

.price-card.highlighted .price-name {
  @apply border-b-0;
}

.price-name {
  @apply p-8 text-3xl font-bold text-center border-b-4;
}

.pricing {
  @apply flex-none mt-auto bg-white rounded-b rounded-t-none overflow-hidden shadow p-6;
}

.price-card.highlighted .price {
  @apply text-4xl text-gray-800;
}

.price {
  @apply w-full pt-6 text-3xl text-gray-600 font-bold text-center;
}

.price span {
  @apply text-base;
}

.price-action {
  @apply flex items-center justify-center;
}

.price-action button {
  @apply mx-auto text-white font-bold rounded-full my-6 py-4 px-8 shadow-lg;
}

.price-action button:hover {
  @apply underline;
}

@screen lg {
  .price-action button {
    @apply mx-0;
  }
}
</style>
