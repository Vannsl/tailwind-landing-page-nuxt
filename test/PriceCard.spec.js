import { shallowMount } from '@vue/test-utils'
import PriceCard from '@/components/PriceCard'

const factory = () => {
  return shallowMount(PriceCard, {
    propsData: {
      name: 'Pro',
      features: ['Thing', 'Thing', 'Thing', 'Thing'],
      price: '29,99 €',
      priceLabel: 'for one user',
      highlighted: true
    }
  })
}

describe('PriceCard', () => {
  it('is a Vue instance', () => {
    const wrapper = factory()
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('renders properly', () => {
    const wrapper = factory()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('has the property name', () => {
    const wrapper = factory()
    expect(wrapper.props('name')).toBe('Pro')
  })

  it('has the property features', () => {
    const wrapper = factory()
    expect(wrapper.props('features')).toEqual([
      'Thing',
      'Thing',
      'Thing',
      'Thing'
    ])
  })

  it('has the property price', () => {
    const wrapper = factory()
    expect(wrapper.props('price')).toBe('29,99 €')
  })

  it('has the property priceLabel', () => {
    const wrapper = factory()
    expect(wrapper.props('priceLabel')).toBe('for one user')
  })

  it('has the property highlighted', () => {
    const wrapper = factory()
    expect(wrapper.props('highlighted')).toBe(true)
  })
})
