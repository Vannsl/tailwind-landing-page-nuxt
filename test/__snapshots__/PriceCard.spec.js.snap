// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceCard renders properly 1`] = `
"<div class=\\"price-card highlighted\\"><div class=\\"price-content\\"><div class=\\"price-name\\">Pro</div> <div class=\\"seperator gradient\\"></div> <ul><li>Thing</li><li>Thing</li><li>Thing</li><li>Thing</li></ul></div> <div class=\\"pricing\\"><div class=\\"price\\">
      29,99 â‚¬
      <span>for one user</span></div> <div class=\\"price-action\\"><button class=\\"gradient\\">Sign Up</button></div></div></div>"
`;
